test/integration/join_cluster_test.js

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var _ = require('underscore');
+var after = require('after');
+var DebuglogLogger = require('../../lib/debug-log-logger.js');
+var Ringpop = require('../../index.js');
+var TChannel = require('tchannel');
+var test = require('tape');
+
+function bootstrapClusterOf(size, opts, tap, onBootstrap) {
```

Comment:
suggestion: move this function and related guys into something like https://github.com/uber/ringpop/blob/master/test/lib/alloc-cluster.js or create a new helper to set up a cluster of ringpops. 

Seems like something you want to write once and re-use for all integration tests.